src/FESpaces.jl

```diff
@@ -11,6 +11,10 @@ function FESpaces.get_free_dof_ids(fs::DistributedFESpace)
   @abstractmethod
 end
 
+function get_dirichlet_dof_ids(f::DistributedFESpace)
```

Comment:
BTW, did you implement this abstract method for the subtypes of DistributedFESpaces? I am not able to see the implementations. If they are not implemented, why arent they required?

Reply:
Fixed in 7030e01a8373acb07da0d5d62a286a79984e0e51.

Regarding the implementation for subtypes, that would only involve `DistributedMultiFieldFESpace`, right? In that case the specialization is not necessary because this function is called for each of the single FE spaces. That's also the case in `Gridap`, where this function is not specialized for `MultiFieldFESpaces`.

Reply:
I have been taken a more close look at the rationale behind the `get_dirichlet_dof_ids` in Gridap.jl. 

The abstract method at hand is defined at the `SingleFieldFESpace` level in the type hierarchy. 

All the subtypes of `SingleFieldFESpace` implement it, as expected 

```
julia> subtypes(Gridap.FESpaces.SingleFieldFESpace)
6-element Vector{Any}:
 Gridap.FESpaces.DirichletFESpace
 Gridap.FESpaces.FESpaceWithConstantFixed
 Gridap.FESpaces.FESpaceWithLinearConstraints
 Gridap.FESpaces.UnconstrainedFESpace
 Gridap.FESpaces.ZeroMeanFESpace
 TrialFESpace
```

The  method is called SOLELY within the whole `Gridap.jl` code at the following two points: 

```julia
num_dirichlet_dofs(f::SingleFieldFESpace) = length(get_dirichlet_dof_ids(f))

function get_free_dof_ids(f::DirichletFESpace)
  get_dirichlet_dof_ids(f.space)
end
```

Going back to GridapDistributed.jl, and given the above, wouldn't it make more sense to define (and implement) the method for `DistributedSingleFieldFESpace`? Should we also implement `num_dirichlet_dofs`?

In any case, I think that before implementing anything we should understand why do we need this method in GridapDistributed. Why do you need it? 



```

Reply:
> Why do you need it?

Ok, I now see that you need it in the following line (sorry, I had missed it):

https://github.com/gridap/GridapDistributed.jl/pull/74/files#diff-8f937e22a7913b08b6cc1767736c1fd408cb88da2cd2ae15a5335f8ca5833fe4R42

> Should we also implement num_dirichlet_dofs?

Ok, I see you have implemented it here:

https://github.com/gridap/GridapDistributed.jl/pull/74/files#diff-8f937e22a7913b08b6cc1767736c1fd408cb88da2cd2ae15a5335f8ca5833fe4R48

Anyway, i still do not get why it is working. Are you actually calling `zero_dirichlet_values(f::DistributedFESpace)` ?